### ISSUE TITLE
on change property for form fields

### ISSUE THREAD

**Issue by NomadicDeveloper22** [ritagroup/gsform#13]
my code is largely based on the example

```
late GSForm form;
...

child: form = GSForm.singleSection(
  ...
```

I use riverpod for state management. Everytime I use the built in flutter showSearch(). My form is reset. 

I would like to capture the form's field's values as they are updated to save their state and restore them once the showSearch() is returned

**Comment by saeeddastras** [comment]
Hello @TheRedSpy15 
To prevent the loss of data, you need to save the values before rebuilding and send them as default values to the fields.

**Comment by saeeddastras** [comment]
Hello @TheRedSpy15 
This  issue has been solved in version 0.3.3 and you can use it by updating the version to 0.3.3
